### PR TITLE
Add UI to filter games by title to home page

### PR DESCRIPTION
**Pull request by magnusrk-ch**
This addresses part of issue #782, and can be extended to support more filters as needed.

Screenshot:

filters

### CODE REVIEW

File: assets/app/view/game_row_filters.rb [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    def render_title(url_search_params)
+      selected_title = url_search_params['title']
+      game_options = [h(:option, { attrs: { value: '' } }, '(All titles)')]
+      Engine::VISIBLE_GAMES.map(&:title).sort.each do |title|
+        # Don't use game::GAME_DROPDOWN_TITLE since it's not what is displayed in game cards.
+        game_options << h(:option, { attrs: { value: title, selected: title == selected_title } }, title)
+      end
+      title_change = lambda do |e|
+        title = e.JS['currentTarget'].JS['value']
+        url_search_params['title'] = title.empty? ? nil : title
```
[the remark this answers is not in the document]
Comment (magnusrk-ch) [reply assets/app/view/game_row_filters.rb:32]: Okay, I suppose the odds of code relying on a distinction between empty value and absent param are low enough.

File: assets/app/view/game_row_filters.rb [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+    def render_reset
+      attrs = {
+        on: { click: -> { update_filters('') } },
+      }
+      h('button', attrs, 'Reset filters')
+    end
+
+    def update_filters(params)
+      params = "?#{params}" unless params.start_with?('?')
```
[the remark this answers is not in the document]
Comment (magnusrk-ch) [reply assets/app/view/game_row_filters.rb:50]: Ack

File: assets/app/view/game_row_filters.rb [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      get_games(params)
+      store(:app_route, "#{@app_route.split('?').first}#{params}")
```
[the remark this answers is not in the document]
Comment (magnusrk-ch) [reply assets/app/view/game_row_filters.rb:52]: Ack

File: assets/app/view/game_row_filters.rb [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+require 'game_manager'
+require 'lib/params'
+
+module View
+  class GameRowFilters < Snabberb::Component
+    include GameManager
+
+    def render
+      url_search_params = Lib::Params::URLSearchParams.new
+      return if url_search_params.unsupported
```
[the remark this answers is not in the document]
Comment (magnusrk-ch) [reply assets/app/view/game_row_filters.rb:12]: Ack

File: models/game.rb [hunk 5]
```diff
@@ -79,13 +81,15 @@ class Game < Base
   # rubocop:enable Style/FormatString
 
   def self.home_games(user, **opts)
+    title = opts['title']
```
[the remark this answers is not in the document]
Comment (magnusrk-ch) [reply models/game.rb:84]: Ack

**Review by tobymao — APPROVED** [review]

**Pull request merged**